src/corelibs/U2Core/src/globals/AutoAnnotationsSupport.h

```diff
@@ -98,6 +98,12 @@ private slots:
 
 #define AUTO_ANNOTATION_SETTINGS "auto-annotations/"
 
+/**
+ * Maximum number of annotations UGENE can handle safely.
+ * Auto-annotation tasks are recommended to avoid producing more annotation than given number to avoid UI freeze.
+ */
+#define AUTO_ANNOTATION_MAX_ANNOTATIONS_ADV_CAN_HANDLE (150 * 1000)
```

Comment:
1. Please, use `static constexpr int AUTO_ANNOTATION_MAX_ANNOTATIONS_ADV_CAN_HANDLE  = 150 * 1000` instead of the ancient `#define`
2. Where did you get this number from? I mean, why this is the maximum?